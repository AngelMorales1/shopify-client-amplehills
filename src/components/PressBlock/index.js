import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import get from 'utils/get';

import { Image } from 'components/base';
import PressCard from 'components/PressCard';

class PressBlock extends Component {
  render() {
    const { model, pressItems, z, setRef, drip } = this.props;

    if (model.isError) return <h1>Error</h1>;

    const pressItemsId = get(pressItems, 'fragments', []).map(fragment =>
      get(fragment[0], 'value', '')
    );
    const firstPressItem = get(
      pressItems,
      `simpleFragments.${pressItemsId[0]}`,
      {}
    );

    return (
      <div
        ref={refBlock => setRef(refBlock)}
        style={{ zIndex: z }}
        className={cx({ drip: drip })}
      >
        <div className="mx-auto my4 px2 col-12 md-col-6 flex flex-column items-center">
          <span className="block-headline text-peach center">{`"${get(
            firstPressItem,
            'quote',
            ''
          )}"`}</span>
          <Image
            className="col-3 mt4 mb2"
            src={get(firstPressItem, 'logoImage.data', '')}
          />
        </div>
        <div className="p3 flex flex-row justify-center flex-wrap">
          {pressItemsId.map((pressItemId, i) => (
            <PressCard
              key={pressItemId}
              pressCard={get(pressItems, `simpleFragments.${pressItemId}`, {})}
            />
          ))}
        </div>
      </div>
    );
  }
}

PressBlock.PropTypes = {
  pressItems: PropTypes.objet,
  z: PropTypes.number,
  setRef: PropTypes.func,
  drip: PropTypes.bool
};

PressBlock.defaultProps = {
  pressItems: {},
  z: 1,
  setRef: () => {},
  drip: false
};

export default PressBlock;
