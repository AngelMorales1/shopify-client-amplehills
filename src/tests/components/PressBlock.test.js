import React from 'react';
import { shallow } from 'enzyme';

import PressBlock from 'components/PressBlock';

it('renders without data', () => {
  const component = shallow(<PressBlock />);

  expect(component).toMatchSnapshot();
});

it('renders with data', () => {
  const component = shallow(
    <PressBlock
      pressItems={{
        simpleFragments: {
          foo: {
            index: 1,
            linkUrl: '/foo',
            logoImage: {
              data: 'foo',
              name: 'foo',
              size: 100,
              type: 'foo'
            },
            quote: 'foo',
            title: 'foo',
            uuid: 'foo'
          }
        }
      }}
      z={1}
      setRef={() => console.log('foo')}
    />
  );

  expect(component).toMatchSnapshot();
});
