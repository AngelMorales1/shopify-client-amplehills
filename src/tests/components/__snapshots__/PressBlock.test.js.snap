// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PressBlock
    drip={false}
    pressItems={
        Object {
            "simpleFragments": Object {
              "foo": Object {
                "index": 1,
                "linkUrl": "/foo",
                "logoImage": Object {
                  "data": "foo",
                  "name": "foo",
                  "size": 100,
                  "type": "foo",
                },
                "quote": "foo",
                "title": "foo",
                "uuid": "foo",
              },
            },
          }
    }
    setRef={[Function]}
    z={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="mx-auto my4 px2 col-12 md-col-6 flex flex-column items-center"
>
          <span
                    className="block-headline text-peach center"
          >
                    ""
          </span>
          <Image
                    alt=""
                    bg={false}
                    className="col-3 mt4 mb2"
                    onImgLoad={[Function]}
                    src=""
                    style={Object {}}
          />
</div>,
        <div
          className="p3 flex flex-row justify-center flex-wrap"
>
          
</div>,
      ],
      "className": "",
      "style": Object {
        "zIndex": 1,
      },
    },
    "ref": [Function],
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="block-headline text-peach center"
>
              ""
</span>,
            <Image
              alt=""
              bg={false}
              className="col-3 mt4 mb2"
              onImgLoad={[Function]}
              src=""
              style={Object {}}
/>,
          ],
          "className": "mx-auto my4 px2 col-12 md-col-6 flex flex-column items-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "\\"\\"",
              "className": "block-headline text-peach center",
            },
            "ref": null,
            "rendered": "\\"\\"",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "",
              "bg": false,
              "children": null,
              "className": "col-3 mt4 mb2",
              "onImgLoad": [Function],
              "src": "",
              "style": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "p3 flex flex-row justify-center flex-wrap",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="mx-auto my4 px2 col-12 md-col-6 flex flex-column items-center"
>
            <span
                        className="block-headline text-peach center"
            >
                        ""
            </span>
            <Image
                        alt=""
                        bg={false}
                        className="col-3 mt4 mb2"
                        onImgLoad={[Function]}
                        src=""
                        style={Object {}}
            />
</div>,
          <div
            className="p3 flex flex-row justify-center flex-wrap"
>
            
</div>,
        ],
        "className": "",
        "style": Object {
          "zIndex": 1,
        },
      },
      "ref": [Function],
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="block-headline text-peach center"
>
                ""
</span>,
              <Image
                alt=""
                bg={false}
                className="col-3 mt4 mb2"
                onImgLoad={[Function]}
                src=""
                style={Object {}}
/>,
            ],
            "className": "mx-auto my4 px2 col-12 md-col-6 flex flex-column items-center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "\\"\\"",
                "className": "block-headline text-peach center",
              },
              "ref": null,
              "rendered": "\\"\\"",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": "",
                "bg": false,
                "children": null,
                "className": "col-3 mt4 mb2",
                "onImgLoad": [Function],
                "src": "",
                "style": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "p3 flex flex-row justify-center flex-wrap",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PressBlock
    drip={false}
    pressItems={Object {}}
    setRef={[Function]}
    z={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="mx-auto my4 px2 col-12 md-col-6 flex flex-column items-center"
>
          <span
                    className="block-headline text-peach center"
          >
                    ""
          </span>
          <Image
                    alt=""
                    bg={false}
                    className="col-3 mt4 mb2"
                    onImgLoad={[Function]}
                    src=""
                    style={Object {}}
          />
</div>,
        <div
          className="p3 flex flex-row justify-center flex-wrap"
>
          
</div>,
      ],
      "className": "",
      "style": Object {
        "zIndex": 1,
      },
    },
    "ref": [Function],
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="block-headline text-peach center"
>
              ""
</span>,
            <Image
              alt=""
              bg={false}
              className="col-3 mt4 mb2"
              onImgLoad={[Function]}
              src=""
              style={Object {}}
/>,
          ],
          "className": "mx-auto my4 px2 col-12 md-col-6 flex flex-column items-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "\\"\\"",
              "className": "block-headline text-peach center",
            },
            "ref": null,
            "rendered": "\\"\\"",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "",
              "bg": false,
              "children": null,
              "className": "col-3 mt4 mb2",
              "onImgLoad": [Function],
              "src": "",
              "style": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "p3 flex flex-row justify-center flex-wrap",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="mx-auto my4 px2 col-12 md-col-6 flex flex-column items-center"
>
            <span
                        className="block-headline text-peach center"
            >
                        ""
            </span>
            <Image
                        alt=""
                        bg={false}
                        className="col-3 mt4 mb2"
                        onImgLoad={[Function]}
                        src=""
                        style={Object {}}
            />
</div>,
          <div
            className="p3 flex flex-row justify-center flex-wrap"
>
            
</div>,
        ],
        "className": "",
        "style": Object {
          "zIndex": 1,
        },
      },
      "ref": [Function],
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="block-headline text-peach center"
>
                ""
</span>,
              <Image
                alt=""
                bg={false}
                className="col-3 mt4 mb2"
                onImgLoad={[Function]}
                src=""
                style={Object {}}
/>,
            ],
            "className": "mx-auto my4 px2 col-12 md-col-6 flex flex-column items-center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "\\"\\"",
                "className": "block-headline text-peach center",
              },
              "ref": null,
              "rendered": "\\"\\"",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": "",
                "bg": false,
                "children": null,
                "className": "col-3 mt4 mb2",
                "onImgLoad": [Function],
                "src": "",
                "style": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "p3 flex flex-row justify-center flex-wrap",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
